catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -407,13 +413,139 @@ protected void printStatus(long count) {
+          .forEach(
+              metacard -> {
+                writeMetacardToZip(zipOutputStream, metacard);
+
+                if (hasLocalResources(metacard)) {
+                  resourceMap.putAll(getAllMetacardContent(metacard));
+                }
+              });
+
+      // write the resources to the zip
+      resourceMap.forEach(
+          (filename, resource) -> writeResourceToZip(zipOutputStream, filename, resource));
+
+    } catch (IOException e) {
+      throw new CatalogTransformerException(
+          String.format(
+              "Error occurred when initializing/closing ZipOutputStream with path %s.", filePath),
+          e);
+    }
+  }
+
+  private void writeMetacardToZip(ZipOutputStream zipOutputStream, Metacard metacard) {
+    InputStream inputStream = null;
+
+    try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(BUFFER_SIZE);
+        ObjectOutputStream objectOutputStream = new ObjectOutputStream(fileBackedOutputStream)) {
+
+      ZipEntry zipEntry = new ZipEntry(METACARD_PATH + metacard.getId());
+      zipOutputStream.putNextEntry(zipEntry);
+
+      objectOutputStream.writeObject(new MetacardImpl(metacard));
+      inputStream = fileBackedOutputStream.asByteSource().openStream();
+
+      IOUtils.copy(inputStream, zipOutputStream);
+    } catch (IOException e) {
+      LOGGER.debug("Failed to add metacard with id {}.", metacard.getId(), e);
+    } finally {
+      try {
+        if (inputStream != null) {
+          inputStream.close();
+        }
+      } catch (IOException e) {
+        LOGGER.debug("Failed to close input stream", e);
+      }
+    }
+  }
+
+  private boolean hasLocalResources(Metacard metacard) {
+    URI uri = metacard.getResourceURI();
+    return (uri != null && ContentItem.CONTENT_SCHEME.equals(uri.getScheme()));
+  }
+
+  private Map<String, Resource> getAllMetacardContent(Metacard metacard) {
+    Map<String, Resource> resourceMap = new HashMap<>();
+    Attribute attribute = metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+
+    if (attribute != null) {
+      List<Serializable> serializables = attribute.getValues();
+      serializables.forEach(
+          serializable -> {
+            String fragment = CONTENT + File.separator;
+            URI uri = null;
+            try {
+              uri = new URI((String) serializable);
+              String derivedResourceFragment = uri.getFragment();
+              if (ContentItem.CONTENT_SCHEME.equals(uri.getScheme())
+                  && StringUtils.isNotBlank(derivedResourceFragment)) {
+                fragment += derivedResourceFragment + File.separator;
+                Resource resource = getResource(metacard);
+                if (resource != null) {
+                  resourceMap.put(
+                      fragment + uri.getSchemeSpecificPart() + "-" + resource.getName(), resource);
+                }
+              }
+            } catch (URISyntaxException e) {
```

Comment:
✏️ If this catch is only for that `new URI(..)` on L 495, it might clarify it some to just wrap that one line that is throwing the error, that way we don't have unnecessary extra code inside of the try block